ompi/mca/coll/acoll/coll_acoll_barrier.c

```diff
@@ -130,21 +130,22 @@ int mca_coll_acoll_barrier_intra(struct ompi_communicator_t *comm, mca_coll_base
     ompi_request_t **reqs;
     int num_nodes;
     mca_coll_acoll_module_t *acoll_module = (mca_coll_acoll_module_t *) module;
-    coll_acoll_subcomms_t *subc;
-    int cid = ompi_comm_get_local_cid(comm);
+    coll_acoll_subcomms_t *subc = NULL;
 
-    /* Fallback to linear if cid is beyond supported limit */
-    if (cid >= MCA_COLL_ACOLL_MAX_CID) {
+    /* Obtain the subcomms structure */
+    err = check_and_create_subc(comm, acoll_module, &subc);
+
+    /* Fallback to linear if subcomms structure is not obtained */
+    if (subc == NULL) {
```

Comment:
minor nit: the Open MPI coding standard require to have the constant as the first argument of the comparison, so this should be  

if (NULL == subc) {

Reply:
Done